docs/website/11ty/resolve-path.transform.js

```diff
@@ -26,11 +27,38 @@ class PathResolver {
     return window.document.documentElement.innerHTML;
   }
 
+  static isValidUrl(urlString) {
+		let url;
+
+		try { 
+	    url = new URL(urlString); 
+	  }
+	  catch(e) { 
+	    return false; 
+	  }
+
+	  return url.protocol === "http:" || url.protocol === "https:";
+	}
+
   static resolveLinks(dom, filePath) {
-    dom.querySelectorAll('a[href^="."]').forEach((link) => {
-      const resolved = PathResolver.resolveLink(link.href, filePath);
-      if (resolved !== link.href) console.info(blue(`Rewrite link "${link.href}" to "${resolved}"`));
-      link.href = resolved;
+    const domain = new URL(url).hostname;
+
+    dom.querySelectorAll('a[href]').forEach((link) => {
+      if (link.href.startsWith('.')) {
+        const resolved = PathResolver.processRewriteRules(link.href, filePath);
+
+        if (resolved !== link.href) console.info(blue(`Rewrite link "${link.href}" to "${resolved}"`));
+        link.href = resolved;
+      }
+
+      if(PathResolver.isValidUrl(link.href)){
+        const linkDomain = new URL(link.href).hostname;
+      
+        if(domain !== linkDomain) {
+          link.target = "_blank";
+          link.rel = "noopener norefferer";
+        }
+      }
```

Comment:
It looks like this logic is needs to be moved to `resolveLink` method or separated between 2 methods
`resolveLink` and `isExternal`

```suggestion
      const resolved = PathResolver.resolveLink(link.href, filePath);
      if (resolved !== link.href) console.info(blue(`Rewrite link "${link.href}" to "${resolved}"`));
      link.href = resolved;

      if (PathResolver.isExternal(link.href)) {
          link.target = '_blank';
          link.rel = 'noopener norefferer';
      }
```

Reply:
Cosmetic: it also looks like we need to extend eslint to validate this files :) There are a lot of small style inconsistencies (spaces, new lines, double quotes, etc).

Reply:
Splitting logic in those methods is the clearest solution, thanks for the suggestion.
Code style is all good now, fixed linter in my IDE